Assets/MixedRealityToolkit.Tests/PlayModeTests/InputEventSystemTests.cs

```diff
@@ -0,0 +1,300 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+#if !WINDOWS_UWP
+// When the .NET scripting backend is enabled and C# projects are built
+// Unity doesn't include the required assemblies (i.e. the ones below).
+// Given that the .NET backend is deprecated by Unity at this point it's we have
+// to work around this on our end.
+using Microsoft.MixedReality.Toolkit.Diagnostics;
+using Microsoft.MixedReality.Toolkit.Input;
+using Microsoft.MixedReality.Toolkit.Utilities;
+using NUnit.Framework;
+using System.Collections;
+using System.Linq;
+using UnityEngine;
+using UnityEngine.TestTools;
+
+namespace Microsoft.MixedReality.Toolkit.Tests
+{
+    class InputEventSystemTests
+    {
+        // Can't use NUnit setup routine, because needs to skip several frames while initializing/removing objects.
+        public IEnumerator SetupMrtkWithoutCursors()
+        {
+            TestUtilities.InitializeMixedRealityToolkitAndCreateScenes(true);
+            TestUtilities.InitializePlayspace();
+
+            IMixedRealityInputSystem inputSystem = null;
+            MixedRealityServiceRegistry.TryGetService(out inputSystem);
+
+            Assert.IsNotNull(inputSystem, "Input system must be initialized");
+
+            // Let input system to register all cursors and managers.
+            yield return null;
+
+            // Switch off / Destroy all input components, which listen to global events
+            Object.Destroy(inputSystem.GazeProvider.GazeCursor as Behaviour);
```

Comment:
For future reference, would it be possible to have a more deterministic disabling of this (i.e. a test hook for killing off global event listeners). The reason I ask is, this is currently exhaustive of the set of global event listeners, but if we were to add another it would be easy to miss adding it here (and it's likely that unexpected bugs could be difficult to diagnose - i.e. we eventually find out that oh, we missed another input listener here, but it took a while to figure out)

Reply:
I was considering to move this part into a utility function.
It shouldn't be confusing, because it asserts in the end, that not all global input handlers are removed. When someone adds a new global handler, they should immediately hit that.